### ISSUE TITLE
WPF Blazor cannot be started at boot. Error: There is no content at 

### ISSUE THREAD

**Issue by JiuLing-zhang** [dotnet/maui#19359]
### Description

I have created a WPF Blazor via the [MS sample ](https://learn.microsoft.com/en/aspnet/core/blazor/hybrid/tutorials/wpf?view=aspnetcore-7.0), but it doesn't be started at boot.

Microsoft.AspNetCore.Components.WebView.Wpf is used in the example, which points to this repository in nuget.

### Steps to Reproduce

1. Clone https://github.com/JiuLing-zhang/TestAutostart.git
2. Publish project. (dotnet publish -c Release -r win-x64 -p:PublishReadyToRun=true --self-contained false)
3. Add startup via registry. (计算机\HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Run)
![regedit.png](https://s2.loli.net/2023/12/12/HR3O7byaMKIVZmt.png)  
4. Restart computer.
5. The application error.
![AutostartError.png](https://s2.loli.net/2023/12/12/ME7vcw9xO2onhfS.png)  

### Link to public reproduction project repository

https://github.com/JiuLing-zhang/TestAutostart.git

### Version with bug

7.0.101

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

Windows 11

**Comment by Eilon** [comment]
Hi @JiuLing-zhang , this could be due to the 'current directory' of the app being the `%system32%` (or similar) folder, which happens with autostart apps. When you launch an app from Explorer or the Start Menu the 'current directory' is normally set to the folder containing the app.

However, the BlazorWebView control does try to force loading its content from wherever the app is _on disk_ so the current directory _shouldn't_ matter. This is the code that does that: https://github.com/dotnet/maui/blob/main/src/BlazorWebView/src/Wpf/BlazorWebView.cs#L258

Can you confirm that all of the app's file are actually on disk in the expected location? That is, go to wherever the published EXE is on disk and check that the `wwwroot` and related files/folders are all there?

**Comment by JiuLing-zhang** [comment]
> Hi @JiuLing-zhang , this could be due to the 'current directory' of the app being the `%system32%` (or similar) folder, which happens with autostart apps. When you launch an app from Explorer or the Start Menu the 'current directory' is normally set to the folder containing the app.
> 
> However, the BlazorWebView control does try to force loading its content from wherever the app is _on disk_ so the current directory _shouldn't_ matter. This is the code that does that: https://github.com/dotnet/maui/blob/main/src/BlazorWebView/src/Wpf/BlazorWebView.cs#L258
> 
> Can you confirm that all of the app's file are actually on disk in the expected location? That is, go to wherever the published EXE is on disk and check that the `wwwroot` and related files/folders are all there?

Hi @Eilon , all documents are complete. I guess it maybe about the path, because the same problem occurred when I started it with cmd.

1. It can run normally through exe.
2. cmd can run normally using the current directory.
3. cmd cannot run when using other directories.

I recorded a gif for this:  
![ErrorGif.gif](https://s2.loli.net/2023/12/13/QmRU5BF3WJ7Mn6A.gif)

**Comment by Eilon** [comment]
@JiuLing-zhang thank you for confirming, this does look like a bug.

Could you try adding this line at the start of your app's `Program.cs` file?

```c#
Environment.CurrentDirectory = Path.GetDirectoryName(typeof(Program).Assembly.Location);
```

If that works, it should be a good workaround for you to use until we can investigate further.

**Comment by JiuLing-zhang** [comment]
> @JiuLing-zhang thank you for confirming, this does look like a bug.
> 
> Could you try adding this line at the start of your app's file?`Program.cs`
> 
> ```cs
> Environment.CurrentDirectory = Path.GetDirectoryName(typeof(Program).Assembly.Location);
> ```
> 
> If that works, it should be a good workaround for you to use until we can investigate further.

Hi @Eilon ,

This path is the real path during build, it won't work on other computers. My project type of WPF.
```c#
// use App, not Program
Path.GetDirectoryName(typeof(App).Assembly.Location);
```
I temporarily solved the problem by doing the following:
```c#
Environment.CurrentDirectory = Path.GetDirectoryName(Process.GetCurrentProcess().MainModule.FileName);
```

Thanks.

**Comment by JiuLing-zhang** [comment]
> > @JiuLing-zhang thank you for confirming, this does look like a bug.
> > Could you try adding this line at the start of your app's file?`Program.cs`
> > ```cs
> > Environment.CurrentDirectory = Path.GetDirectoryName(typeof(Program).Assembly.Location);
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > If that works, it should be a good workaround for you to use until we can investigate further.
> 
> Hi @Eilon ,
> 
> This path is the real path during build, it won't work on other computers. My project type of WPF.
> 
> ```cs
> // use App, not Program
> Path.GetDirectoryName(typeof(App).Assembly.Location);
> ```
> 
> I temporarily solved the problem by doing the following:
> 
> ```cs
> Environment.CurrentDirectory = Path.GetDirectoryName(Process.GetCurrentProcess().MainModule.FileName);
> ```
> 
> Thanks.

I made a mistake, `typeof(App).Assembly.Location` takes effect at runtime, but the value is null, so it's not work.

**Comment by Eilon** [comment]
@JiuLing-zhang oh interesting. Perhaps because of the publish/ReadyToRun, that causes the assembly to no longer exist in its original form, so that ends up being null. That is probably the root cause of this issue. I think that gives us enough info to try to come up with a more permanent solution.

Does the workaround always work in your case?

**Comment by JiuLing-zhang** [comment]
@Eilon I changed some computers for test and it works.
```c#
Environment.CurrentDirectory = Path.GetDirectoryName(Process.GetCurrentProcess().MainModule.FileName);
```

**Comment by Mr-Technician** [comment]
I am also running into this issue when publishing a single-file Winforms app with Blazor in Webview. The issue does not occur for a normal (not single-file) build.

**Comment by Eilon** [comment]
> I am also running into this issue when publishing a single-file Winforms app with Blazor in Webview. The issue does not occur for a normal (not single-file) build.

@Mr-Technician does the CurrentDirectory workaround above work for you? It's likely the same issue.

**Comment by Mr-Technician** [comment]
@Eilon This is my current solution:

```csharp
blazorWebView1.HostPage = Path.Combine(Path.GetDirectoryName(Process.GetCurrentProcess().MainModule.FileName), "wwwroot\\index.html");
```

It is essentially the same as setting `Environment.CurrentDirectory`.

**Comment by mkArtakMSFT** [comment]
This was fixed by #21750
